pkg/component/runtime/conn_info_server.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+type connInfoServer struct {
+	listener net.Listener
+	wg       sync.WaitGroup
+}
+
+func newConnInfoServer(comm Communicator, port int, log *logger.Logger) (*connInfoServer, error) {
+	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", port))
+	if err != nil {
+		return nil, fmt.Errorf("failed to start connection credentials listener: %w", err)
+	}
+
+	s := &connInfoServer{listener: listener}
+
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		for {
+			conn, err := listener.Accept()
+			if err != nil {
+				log.Errorf("failed accept conn info connection: %v", err)
```

Comment:
this one is a bit tricky, looking at the current V1 code this service is running all the time on this port.

1. should we always wait until the service connects on start? what if the service is already running?
2. should we set the state to "failed/degraded" if this comms failed? even though the service can reconnect to the agent later and succeed? as far as I can tell from V1 implementation the service could connect to the creds port at any time (like for example after service restart).
3. now it seems like we are talking about merging the states from different sources cause they run at the same time, need to track the source of failure in the runtime: 1. service runtime primary loop and 2. the connection info server loop. 
let say the service runtime primary loop failed and set to the state to failed, and then later we get the "success" state, so depending on the source we have two cases:
a) if we get the success state from the primary loop and the primary loop was the source of last failure then we can set the state to healthy
b) if we get the success state from the connection info server, but the source of the failure was the primary service runtime loop (check in failure), we can't set the state back to healthy.

4. if the connection info server failed once , but the service main loop is healthy and we receive check ins from the service. are we in the healthy or degraded state? should we wait for another connection to the connection info server in order to switch from the degraded to the healthy state? what if we service will never connect, and will be working just fine? 
we could potentially have a quick transitioning state failures in these case that will not be captured on the server because the checkins are not happening that often and log would be the only source to see this error. 

just want to make sure I understand the requirements for this particular part. @blakerouse  could you please comment?